server/cluster_worker.go

```diff
@@ -113,3 +115,41 @@ func (c *RaftCluster) handleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSp
 
 	return split, nil
 }
+
+func (c *RaftCluster) checkSplitRegion(left *metapb.Region, right *metapb.Region) error {
+	if left == nil || right == nil {
+		return errors.New("invalid split region")
+	}
+
+	if !bytes.Equal(left.GetEndKey(), right.GetStartKey()) {
+		return errors.New("invalid split region")
+	}
+
+	if bytes.Compare(left.GetStartKey(), right.GetEndKey()) >= 0 {
+		return errors.New("invalid split region")
+	}
+
+	return nil
+}
+
+func (c *RaftCluster) handleReportSplit(request *pdpb.ReportSplitRequest) (*pdpb.ReportSplitResponse, error) {
+	left := request.GetLeft()
+	right := request.GetRight()
+
+	err := c.checkSplitRegion(left, right)
+	if err != nil {
+		log.Warnf("report split region is invalid - %v, %v", request, errors.ErrorStack(err))
+		return nil, errors.Trace(err)
+	}
+
+	// Build origin region by using left and right.
+	originRegion := cloneRegion(left)
```

Comment:
It will not use epoch i think, it only cares the origin region and split regions info, like starkey/endkey and peers.